utils/js/LangUtils.js

```diff
@@ -30,9 +30,14 @@ mixin(_inflection);
  * @returns {*} value stored at key
  */
 export function getOrCreate(obj, key, fn) {
-    const val = obj[key];
-    if (!isUndefined(val)) return val;
-    return obj[key] = fn();
+    if (obj instanceof Map || obj instanceof WeakMap) {
+        if (obj.has(key)) return obj.get(key);
```

Comment:
Can we use use the same logic we use in bottom branch to use just one get() for both cache hits and misses?

Probably not hugely important performance wise, but probably better and also good to be consistent with what we already do.